test/e2e/feature/feature.go

```diff
@@ -0,0 +1,132 @@
+	SELinux                            = framework.WithFeature(framework.ValidFeatures.Add("SELinux"))
+	SELinuxMountReadWriteOncePod       = framework.WithFeature(framework.ValidFeatures.Add("SELinuxMountReadWriteOncePod"))
+	StackdriverAcceleratorMonitoring   = framework.WithFeature(framework.ValidFeatures.Add("StackdriverAcceleratorMonitoring"))
+	StackdriverCustomMetrics           = framework.WithFeature(framework.ValidFeatures.Add("StackdriverCustomMetrics"))
+	StackdriverExternalMetrics         = framework.WithFeature(framework.ValidFeatures.Add("StackdriverExternalMetrics"))
+	StackdriverMetadataAgent           = framework.WithFeature(framework.ValidFeatures.Add("StackdriverMetadataAgent"))
+	StackdriverMonitoring              = framework.WithFeature(framework.ValidFeatures.Add("StackdriverMonitoring"))
+	StandaloneMode                     = framework.WithFeature(framework.ValidFeatures.Add("StandaloneMode"))
+	StatefulSet                        = framework.WithFeature(framework.ValidFeatures.Add("StatefulSet"))
+	StatefulSetStartOrdinal            = framework.WithFeature(framework.ValidFeatures.Add("StatefulSetStartOrdinal"))
+	StatefulUpgrade                    = framework.WithFeature(framework.ValidFeatures.Add("StatefulUpgrade"))
+	StorageProvider                    = framework.WithFeature(framework.ValidFeatures.Add("StorageProvider"))
+	StorageVersionAPI                  = framework.WithFeature(framework.ValidFeatures.Add("StorageVersionAPI"))
+	TopologyHints                      = framework.WithFeature(framework.ValidFeatures.Add("Topology Hints"))
+	UDP                                = framework.WithFeature(framework.ValidFeatures.Add("UDP"))
+	Upgrade                            = framework.WithFeature(framework.ValidFeatures.Add("Upgrade"))
+	UserNamespacesStatelessPodsSupport = framework.WithFeature(framework.ValidFeatures.Add("UserNamespacesStatelessPodsSupport"))
+	ValidatingAdmissionPolicy          = framework.WithFeature(framework.ValidFeatures.Add("ValidatingAdmissionPolicy"))
+	Volumes                            = framework.WithFeature(framework.ValidFeatures.Add("Volumes"))
+	VolumeSnapshotDataSource           = framework.WithFeature(framework.ValidFeatures.Add("VolumeSnapshotDataSource"))
+	VolumeSourceXFS                    = framework.WithFeature(framework.ValidFeatures.Add("VolumeSourceXFS"))
+	Vsphere                            = framework.WithFeature(framework.ValidFeatures.Add("vsphere"))
+	WatchList                          = framework.WithFeature(framework.ValidFeatures.Add("WatchList"))
+	Windows                            = framework.WithFeature(framework.ValidFeatures.Add("Windows"))
+	WindowsHostProcessContainers       = framework.WithFeature(framework.ValidFeatures.Add("WindowsHostProcessContainers"))
+	WindowsHyperVContainers            = framework.WithFeature(framework.ValidFeatures.Add("WindowsHyperVContainers"))
+)
+
+func init() {
+	// This prevents adding additional ad-hoc features in tests.
```

Comment:
didn't have time to go through the whole PR logic, do you mind explain how this prevents that?

Reply:
Once this package is done with registering the pre-defined features, it locks the registry. Tests that import this package then cannot add more of them.

There is a loophole: a test that doesn't import the package could define features during its own init and, depending on the order in which packages get initialized, might succeed with that. This is only a problem for init. Describe callbacks get invoked after the global init, so something like this will always fail:
```
_ = Describe("foo", func() {
   It("bar", framework.WithFeature("my-custom-feature"), func() { ...})
})
```

We can also drop this locking and rely instead on code reviews to prevent a proliferation of arbitrary feature tags. I'm open for suggestions here.

Reply:
everything that is not automated is doomed to fail XD